src/main/java/tutorlink/storage/GradeStorage.java

```diff
@@ -0,0 +1,70 @@
+package tutorlink.storage;
+
+import tutorlink.component.Component;
+import tutorlink.grade.Grade;
+import tutorlink.student.Student;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class GradeStorage extends Storage {
+    private final ArrayList<Component> componentList;
+    private final ArrayList<Student> studentList;
+
+    public GradeStorage(String filePath, ArrayList<Component> componentList, ArrayList<Student> studentList) {
+        super(filePath);
+        this.componentList = componentList;
+        this.studentList = studentList;
+    }
+
+    public ArrayList<Grade> loadGradeList()
+            throws IOException {
+        ArrayList<Grade> grades = new ArrayList<>();
+        Scanner fileScanner = new Scanner(path);
+        while (fileScanner.hasNext()) {
+            String[] stringParts = fileScanner.nextLine().split(READ_DELIMITER);
+            String componentName = stringParts[0];
+            String matricNumber = stringParts[1];
+            double score = Double.parseDouble(stringParts[2]);
+
+            Component selectedComp = null;
+            for (Component comp : componentList) {
+                if (comp.getName().equals(componentName)) {
+                    selectedComp = comp;
+                    break;
+                }
+            }
+            Student selectedStudent = null;
+            for (Student student : studentList) {
+                if (student.getMatricNumber().equals(matricNumber)) {
+                    selectedStudent = student;
+                    break;
+                }
+            }
+
+            if (selectedComp != null && selectedStudent != null) {
+                Grade newGrade = new Grade(selectedComp, selectedStudent, score);
+                grades.add(newGrade);
+            }
```

Comment:
I like how you check that for each grade in the gradelist, the referenced student and component exists in their respective component lists. 

We will have to explain this in our UG.